helix-core/src/main/java/org/apache/helix/controller/dataproviders/BaseControllerDataProvider.java

```diff
@@ -96,13 +99,15 @@
 
   // Special caches
   private CurrentStateCache _currentStateCache;
+  private CustomizedStateCache _customizedStateCache;
```

Comment:
We aggregate _currentStateCache in addition to externalView because it is used in the rebalance pipeline. For customize state though, the customized view is the only thing we need. Can we just combine these 2 refresh workflow and keep one customized View map in the cache?

Reply:
So _customizedStateCache is used by CustomizedStateComputation stage, which happened before customized view aggregation stage. Therefore, we do need this cache to avoid access to ZK during computation.

Reply:
Have moved it to resource provider.